lib/api/api-request.js

```diff
@@ -153,6 +153,33 @@ class RequestHandler extends AsyncResource {
   }
 }
 
+function getResolveErrorBodyCallback ({ callback, body, contentType, statusCode, statusMessage, headers }) {
+  if (statusCode === 204 || !contentType) {
+    body.dump()
+    callback(new ResponseStatusCodeError(`Response status code ${statusCode}${statusMessage ? `: ${statusMessage}` : ''}`, statusCode, headers))
+    return
+  }
+
+  if (contentType.startsWith('application/json')) {
+    body.json()
+      .then((payload) => {
+        callback(new ResponseStatusCodeError(`Response status code ${statusCode}${statusMessage ? `: ${statusMessage}` : ''}`, statusCode, headers, payload))
+      })
+    return
+  }
+
+  if (contentType.startsWith('text/')) {
+    body.text()
+      .then((payload) => {
+        callback(new ResponseStatusCodeError(`Response status code ${statusCode}${statusMessage ? `: ${statusMessage}` : ''}`, statusCode, headers, payload))
+      })
+    return
+  }
+
+  body.dump()
+  callback(new ResponseStatusCodeError(`Response status code ${statusCode}${statusMessage ? `: ${statusMessage}` : ''}`, statusCode, headers))
+}
```

Comment:
Just make the function async... it's easier

Reply:
done